src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -3751,8 +3751,9 @@ void MatterDoorLockClusterServerAttributeChangedCallback(const app::ConcreteAttr
 // Timer callbacks
 // =============================================================================
 
-void emberAfPluginDoorLockOnAutoRelock(chip::EndpointId endpointId)
+void __attribute__((weak)) emberAfPluginDoorLockOnAutoRelock(chip::EndpointId endpointId)
```

Comment:
Wait, why is this needed?  Shouldn't there be a callback into the app-provided delegate from `DoorLockServer::Instance().SetLockState` here if the app needs to be informed this is happening?  If there is no such callback, that sounds like the real problem....

Reply:
I was trying to maintain how the `door-lock-server.h` seems to be implemented, which is a series of callback stubs for command handling to be implemented by the application. There is no direct callback for the SetLockState if that is what you ask or at least I don't see it. I could use the `MatterPostAttributeChangeCallback` to reflect the lock state change in the app but that is not how it was implemented for any of the other events that change the door lock attributes.

I thought `emberAfPluginDoorLockOnAutoRelock` should be a callback that can be implemented by the app like all the other stubs listed in `src/app/clusters/door-lock-server/door-lock-server-callback.cpp`

Reply:
You're right; I had forgotten how door lock is set up.

If we follow how the rest of door lock works, there should be two functions here:

1. Some function (name TBD) that's set up as the timer handler and does what `emberAfPluginDoorLockOnAutoRelock does right now and calls `emberAfPluginDoorLockOnAutoRelock`.
2. `emberAfPluginDoorLockOnAutoRelock`, which is a weak symbol in `door-lock-server-callback.cpp` which just returns void and does nothing and can be overridden by the application to do other additional things as needed.

In terms of this PR, what that means is that the "set event control inactive" and "set lock state" bits would stay in the door lock cluster code and the app would just need to do the `LockMgr().InitiateAction(AppEvent::kEventType_Lock, LockManager::LOCK_ACTION);` bit...  Or we could consider moving the `SetLockState` call into the app too, but note that in this PR the app call is getting the `DlOperationSource` wrong, which is a very good argument for not just leaving it to the app...

Reply:
alright, that sounds good. I will update the PR and create this second weak callback for app implementation